css/selectors/focus-visible-007.html

```diff
@@ -10,98 +10,48 @@
-      delete document.body.dataset.hadmousedown;
-      document.body.dataset.hadkeydown = "";
-    }, true);
-
-    document.body.addEventListener("mousedown", (e) => {
-      delete document.body.dataset.hadkeydown;
-      document.body.dataset.hadmousedown = "";
-    }, true);
-
     async_test(function(t) {
-      let tested_modality_change = false;
-      let tested_modality_unchanged_by_mouse_click = false;
-      let tested_mouse_focus_after_modality_change = false;
-
-      // TODO(crbug.com/828858): Remove this check once bug is resolved.
-      test_driver.send_keys(document.body, "\uE050").then(t.step_func(() => {
-        const handle_initial_focus = t.step_func(() => {
-          one.addEventListener("keyup", t.step_func(test_modality_change));
-          one.removeEventListener("focus", handle_initial_focus);
+      target.addEventListener("focus", t.step_func(function() {
+        assert_equals(getComputedStyle(target).outlineColor, "rgb(0, 0, 255)", `outlineColor for ${target.tagName}#${target.id} should be blue`);
 
-          test_driver.send_keys(one, "\uE050");
-        });
-
-        const test_modality_change = t.step_func(() => {
-          assert_equals(getComputedStyle(one).outlineColor, "rgb(59, 153, 252)");
-          tested_modality_change = true;
-          one.removeEventListener("keyup", test_modality_change);
-          one.addEventListener("click", test_modality_unchanged_by_mouse_click);
-          test_driver.click(one);
-        });
-
-        const test_modality_unchanged_by_mouse_click = t.step_func(() => {
-          assert_true(tested_modality_change);
-          assert_equals(getComputedStyle(one).outlineColor, "rgb(59, 153, 252)");
-          tested_modality_unchanged_by_mouse_click = true;
-          one.removeEventListener("click", test_modality_unchanged_by_mouse_click);
-          two.addEventListener("focus", test_mouse_focus_after_modality_change);
-          test_driver.click(two);
-        });
-
-        const test_mouse_focus_after_modality_change = t.step_func(() => {
-          assert_true(tested_modality_unchanged_by_mouse_click);
-          assert_false("hadkeydown" in document.body.dataset);
-          assert_equals(getComputedStyle(two).outlineColor, "rgb(59, 153, 252)");
-          tested_mouse_focus_after_modality_change = true;
-          t.done();
-        });
+        const shift_key = "\uE050";
+        test_driver.send_keys(target, shift_key);
+      }));
 
-        one.addEventListener("focus", handle_initial_focus);
-        test_driver.click(one);
-      })).catch(t.step_func((e) => {
-        // TODO(crbug.com/828858): Remove this check once bug is resolved.
-        assert_true(false, "send_keys not implemented yet");
+      target.addEventListener("keyup", t.step_func(function() {
+        assert_equals(getComputedStyle(target).outlineColor, "rgb(0, 128, 0)", `outlineColor for ${target.tagName}#${target.id} should be green`);
         t.done();
```

Comment:
Nit: I think you can replace the `t.step_func` with `t.step_func_done` and then remove this `t.done()`.

Reply:
Done.